libs/types/src/election.ts

```diff
@@ -121,13 +116,18 @@ export interface Candidate {
   readonly partyId?: PartyId;
   readonly isWriteIn?: boolean;
 }
-export const CandidateSchema: z.ZodSchema<Candidate> = z.object({
-  _lang: TranslationsSchema.optional(),
-  id: CandidateIdSchema,
-  name: z.string().nonempty(),
-  partyId: PartyIdSchema.optional(),
-  isWriteIn: z.boolean().optional(),
-});
+export const CandidateSchema: z.ZodSchema<Candidate> = z
+  .object({
+    _lang: TranslationsSchema.optional(),
+    id: CandidateIdSchema,
+    name: z.string().nonempty(),
+    partyId: PartyIdSchema.optional(),
+    isWriteIn: z.boolean().optional(),
+  })
+  .refine(
+    ({ id, isWriteIn }) => isWriteIn || !id.startsWith('write-in'),
```

Comment:
Seems like this should be `isWriteIn === id.startsWith('write-in')`, right? Then it will fail in either wrong case.

Reply:
Hm, I _think_ my test cases cover both cases, and this change makes those tests fail.

Reply:
Good catch! Added another test case and fixed in 0b000fee7d7090dfb95ae949ed46144ed4910666